dashboard/src/main/java/com/google/cloud/tools/opensource/dashboard/DashboardMain.java

```diff
@@ -405,4 +414,44 @@ public static long countFailures(List<ArtifactResults> table, String columnName)
         .filter(row -> row.getResult(columnName) == null || row.getFailureCount(columnName) > 0)
         .count();
   }
+
+  /**
+   * Returns mapping from jar file to simplified description of its {@link DependencyPath}s. A
+   * simplified description of {@link DependencyPath}s explains a common pattern ({@code
+   * groupId:artifactId}) in the path elements, helping to avoid listing more than 5 repetitive
+   * items in BOM dashboard.
+   */
+  private static ImmutableMap<String, String> simplifyDependencyPathMessage(
+      Multimap<Path, DependencyPath> jarToDependencyPaths) {
+    // Freemarker is not good at handling non-string key
+    // https://freemarker.apache.org/docs/app_faq.html#faq_nonstring_keys
+    ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+
+    for (Path jar : jarToDependencyPaths.keySet()) {
+      List<Set<String>> commonVersionLessCoordinates = Lists.newArrayList();
+      Collection<DependencyPath> dependencyPaths = jarToDependencyPaths.get(jar);
+      for (DependencyPath dependencyPath : dependencyPaths) {
+        // Set of "groupId:artifactId"
+        Set<String> versionLessCoordinates =
+            dependencyPath.getPath().stream().map(Artifacts::makeKey).collect(Collectors.toSet());
+        commonVersionLessCoordinates.add(versionLessCoordinates);
+      }
+      Set<String> intersection = Sets.newHashSet(Iterables.getFirst(commonVersionLessCoordinates,
+          Sets.newHashSet()));
+      for (Set<String> versionLessCoordinates : commonVersionLessCoordinates) {
+        intersection.retainAll(versionLessCoordinates);
```

Comment:
This looks wrong. Only elements in the last pass through the loop will be retained.

Reply:
Simplified the logic; no first pass or last pass any more.